### ISSUE TITLE
Trying to learn how you publish on Nuget

### ISSUE THREAD

**Issue by oooolivierrrr** [lecaillon/Evolve#113]
By using a azure artifacts. I used the build definition you published (with a modification).

`pool:
  name: Hosted Ubuntu 1604`

I also added at the end.

`
- task: DotNetCoreCLI@2
  inputs:
    command: 'pack'
    configuration: 'release'
    packagesToPack: 'src/**/*.csproj'
    verbosityPack: 'detailed'

- task: DotNetCoreCLI@2
  inputs:
    command: 'push'
    nuGetFeedType: 'internal'
    packagesToPush: '$(build.artifactStagingDirectory)/*.nupkg'
    publishVstsFeed: 'MyFeed'`

This changes publish the 3 packages. But when I use the evolve.msbuild in a project, the csproj is not modified and the post build task is not called.

Can you tell me what am I missing ?

Ps : this is for a highly regulated domain business which need to get its own repo

**Comment by lecaillon** [comment]
The Evolve packages are manually published to Nuget. That's why you don't find them in the build. (By the way I just updated it with `pool: name: Hosted Ubuntu 1604`)

Next, I think you don't need the first task I see above. Because the build already pack everything in the `dist` folder

1. Concerning the MSBuild package, after adding a reference to it in your project, you should see that your csproj file has been updated like this (if not, there is a problem somewhere): 

``` xml
<PackageReference Include="Evolve.MSBuild.Windows.x64" Version="2.2.0">
  <PrivateAssets>all</PrivateAssets>
  <IncludeAssets>runtime; build; native; contentfiles; analyzers; buildtransitive</IncludeAssets>
</PackageReference>
```

After added the reference, add an `evolve.json`  or update your `web.config` configuration file like in the [documentation](https://evolve-db.netlify.com/msbuild/).

2. An other option is to use Evolve directly in your [code](https://evolve-db.netlify.com/getting-started/) with the [Evolve package](https://www.nuget.org/packages/Evolve/) (not the MSBuild one). 

3. And finally you can use the [CLI](https://evolve-db.netlify.com/cli/) in your CI pipeline, to deploy your database changes in production.

What ever solution you take, depending your constraints and wishes, you have to put and name your SQL migration scripts following a convention explained [here](https://evolve-db.netlify.com/concepts/) and [here](https://evolve-db.netlify.com/configuration/)

In conclusion solution 1. is to keep uptodate you dev environment, solution 2. for dev and for prod, solution 3. for prod only I would say.

**Comment by oooolivierrrr** [comment]
I am trying to publish on a private feed by using a build definition on Devops.

Up to now, I am able to publish everything on the private feed. But when I am trying to use it, it does not automatically set up the .csproj like your package on nuget does on its own.

When I use the private feed, the csproj has the correct reference but the PrivateAssets and IncludeAssets sections are missing. I added those section myself and runned a build : usually the output of visual studio  states that Evolve is starting but nothing but a normal build was outputed. 

I uninstalled the package and used the yours instead, still in the same project.
And now, the .csproj is correctly modified and on a build, Evolve is started and correctly applied the migrations files. I checked the database and the migrations are applied by the msbuild post build task.

I don't see what's happening.

**Comment by lecaillon** [comment]
If you do not have those sections, it is like I said, because you do not use the official packages, that are already packed in the `dist` folder during the build. While I can understand that you want to publish on a private feed I do not, when you pack your own package... Use those already built in the `dist` folder, is the best option for you.
So remove your first VSTS task and update the second like something this:
``` yml
task: DotNetCoreCLI@2
inputs:
command: 'push'
nuGetFeedType: 'internal'
packagesToPush: 'dist/*.nupkg'
publishVstsFeed: 'MyFeed'
```

**Comment by oooolivierrrr** [comment]
I did add the pack command because I did not see the packages in the dist folder. I think the yaml produced for azure does not pack them.
I did that before I wrote my answer : I forgot to mention it.  Even without the pack command the azure_pipelines.yaml does not produce the nupkg.
I am trying to write an yaml which replicate the appveyor one because I think this one is calling the cake build which in return produces the packages.

**Comment by lecaillon** [comment]
You are right. Indeed the MSBuild package is Windows platform dependant and so can only be built in the Windows `AppVeyor` CI, not the Linux `VSTS` one. You can not build this package on Linux because it needs the Windows CLI.

But like I said before, you don't need the Evolve.MSBuild.Windows.x64 package to use Evolve. In your case I would only use the Evolve package with maybe the CLI to apply your migrations in prod (or only the Evolve package).

**Comment by oooolivierrrr** [comment]
I though the MSBuild package can be build on Linux since it supports .Net Core (or at least .NetStandard). Wrong assumption from my part : I'll dig some more on MSBuild projects.

You're right I can kickstart by using the Evolve package and the CLI in prod. This is probably what I'll do until I manage to build on azure.

MSBuild seems not necessary but I want migrations application to be an operational matter (I don't want devs to be bothered by it).

**Comment by oooolivierrrr** [comment]
That's why I want it so bad !

**Comment by lecaillon** [comment]
Those few lines in a method called at the startup of your application, should be acceptable:
``` c#
try
{
    var cnx = new SqliteConnection(Configuration.GetConnectionString("MyDatabase"));
    var evolve = new Evolve.Evolve(cnx, msg => _logger.LogInformation(msg))
    {
        Locations = new[] { "db/migrations" },
        IsEraseDisabled = true,
    };

    evolve.Migrate();
}
catch (Exception ex)
{
    _logger.LogCritical("Database migration failed.", ex);
    throw;
}
```

With that, using the CLI is not necessary. Just use Evolve in the code.

**Comment by oooolivierrrr** [comment]
Going to follow this way. Thanks for your input.
